ngraph/core/reference/include/ngraph/runtime/reference/convolution_backprop_data.hpp

```diff
@@ -23,217 +19,224 @@ namespace ngraph
+                        input_3d[input_3d.size() - strides.size() + i] = input_shape[i + 2];
+                        strides_3d[strides_3d.size() - strides.size() + i] = strides[i];
+                        new_input_3d[new_input_3d.size() - strides.size() + i] =
+                            output_shape[i + 2];
                     }
 
-                    AxisVector in_transform_axis_order(2 + n_spatial_dimensions);
-                    for (size_t i = 0; i < in_transform_axis_order.size(); i++)
+                    const size_t input_size = shape_size(input_3d);
+                    if (input_size == 1)
                     {
-                        in_transform_axis_order[i] = i;
+                        for (size_t i = 0; i < shape_size(input_shape); ++i)
+                        {
+                            input_zeros.push_back(in[i]);
+                        }
                     }
-                    CoordinateTransform in_transform(in_shape,
-                                                     in_transform_start,
-                                                     in_transform_end,
-                                                     in_transform_movement_strides,
-                                                     in_transform_axis_order,
-                                                     in_transform_pad_below,
-                                                     in_transform_pad_above,
-                                                     in_transform_dilation_strides);
-
-                    // Simultaneously with iterating I, for the filter we need to iterate the
-                    // coordinate:
-                    //
-                    //   F
-                    //
-                    // over the range (noninclusive on the right):
-                    //
-                    //   (chan_out,0,0,...,0) ->
-                    //     (chan_out+1,
-                    //      chans_in_count,
-                    //      filter_dims_1,
-                    //        ...,
-                    //      filter_dims_n)
-                    //
-                    // with unit stride.
-
-                    Shape filter_transform_start(2 + n_spatial_dimensions);
-                    Shape filter_transform_end(2 + n_spatial_dimensions);
-
-                    filter_transform_start[filter_out_channel_axis] = out_channel;
-                    filter_transform_end[filter_out_channel_axis] = out_channel + 1;
-                    filter_transform_start[filter_in_channel_axis] = 0;
-                    filter_transform_end[filter_in_channel_axis] = 1;
-
-                    for (size_t i = 2; i < n_spatial_dimensions + 2; i++)
+                    else
                     {
-                        filter_transform_start[i] = 0;
-                        filter_transform_end[i] = filter_shape[i];
+                        for (size_t batch = 0; batch < input_shape[0]; ++batch)
```

Comment:
```suggestion
                        for (size_t batch = 0; batch < input_shape[0]; ++batch)
                        {
                             const auto offset_batch = batch * input_size * input_shape[1];
                             for (size_t channel = 0; channel < input_shape[1]; ++channel)
                             {
                                 const auto offset_channel = offset_batch + channel * input_size;
```
Etc for `i_z`, `i_y` and `i_x`

I think that it is more optimal

Reply:
Applied suggested proposition.